Lib/test/test_zipfile/_path/test_path.py

```diff
@@ -454,9 +454,15 @@ def test_root_unnamed(self, alpharep):
 
         # .name and .parent should still work on subs
         sub = root / "b"
-        assert sub.name == "b"
+        self.assertEqual(sub.name, "b")
         assert sub.parent
 
+        # It should be possible to get the repr and str
+        self.assertEqual(str(root), ":memory:/")
+        self.assertEqual(repr(root), "Path(None, '')")
+        self.assertEqual(str(sub), ":memory:/b/")
+        self.assertEqual(repr(sub), "Path(None, 'b/')")
+
```

Comment:
This looks like a separate test. Let's make it so, and give it a docstring that explains the expectation (something like ```In-memory Path has a valid ``str`` and ``repr``. ```) and links to the issue (`python/cpython#103120`).